### ISSUE TITLE
Install of 23.1 not right ?

### ISSUE THREAD

**Issue by markhooper99** [darold/ora2pg#1444]
I installed version 23.1 over our existing version 23.0 today because I'm hoping that there's a fix to recognize the reserved word 'partition'.  
However, I noticed the 'make' process didn't replace the /usr/local/bin/ora2pg and /usr/local/bin/ora2pg_scanner files as it is supposed to do (I believe).

Here's the install ...

```
hou-db6:root:testhou> perl Makefile.PL
Checking if your kit is complete...
Looks good
Invalid LICENSE value 'GPLv3' ignored
Generating a Unix-style Makefile
Writing Makefile for Ora2Pg
Invalid LICENSE value 'GPLv3' ignored
Writing MYMETA.yml and MYMETA.json

Done...
------------------------------------------------------------------------------
Please read documentation at http://ora2pg.darold.net/ before asking for help
------------------------------------------------------------------------------
Now type: make && make install
hou-db6:root:testhou> make && make install
cp lib/Ora2Pg/GEOM.pm blib/lib/Ora2Pg/GEOM.pm
cp lib/Ora2Pg.pm blib/lib/Ora2Pg.pm
cp lib/Ora2Pg/PLSQL.pm blib/lib/Ora2Pg/PLSQL.pm
cp lib/Ora2Pg/Oracle.pm blib/lib/Ora2Pg/Oracle.pm
cp lib/Ora2Pg/MySQL.pm blib/lib/Ora2Pg/MySQL.pm
cp scripts/ora2pg blib/script/ora2pg
"/usr/bin/perl" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/ora2pg
cp scripts/ora2pg_scanner blib/script/ora2pg_scanner
"/usr/bin/perl" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/ora2pg_scanner
Manifying 1 pod document
Manifying 1 pod document
Installing /root/perl5/lib/perl5/Ora2Pg.pm
Installing /root/perl5/lib/perl5/Ora2Pg/Oracle.pm
Installing /root/perl5/lib/perl5/Ora2Pg/MySQL.pm
Installing /root/perl5/lib/perl5/Ora2Pg/GEOM.pm
Installing /root/perl5/lib/perl5/Ora2Pg/PLSQL.pm
Installing /root/perl5/man/man3/ora2pg.3
Installing /root/perl5/bin/ora2pg
Installing /root/perl5/bin/ora2pg_scanner
Installing default configuration file (ora2pg.conf.dist) to /etc/ora2pg
Appending installation info to /root/perl5/lib/perl5/x86_64-linux-thread-multi/perllocal.pod
```

Anyways, it looks like it's installed...

```
hou-db6:markh:testhou> ora2pg --version
Ora2Pg v23.1
```

I manually copied /root/perl5/bin/ora2pg to /usr/local/bin and cp /root/perl5/bin/ora2pg_scanner to /usr/local/bin
(I'm sure this isn't supposed to be done, right?).

I added a few things to my ora2pg config file for the word 'partition' ...
```
ORA_RESERVED_WORDS      audit,comment,references,partition
USE_RESERVED_WORDS      1

```

Anyways, when I run ora2pg in a shell script it thinks its version 23.1 (when reported by -version) but it thinks its version 23.0 when trying to actually do a copy from an Oracle table to PG...

Shell script ...
```
#!/bin/sh
ora2pg -version
ora2pg -d -c /tmp/ora2pg_marktest.conf -t copy

```

Run...
```
hou-db6:markh:testhou> ./marktest.sh
Ora2Pg v23.1
[2022-06-22 16:59:25] Ora2Pg version: 23.0
[2022-06-22 16:59:25] Export type: COPY
[2022-06-22 16:59:25] Geometry export type: INTERNAL
[2022-06-22 16:59:25] ORACLE_HOME = /usr/lib/oracle/12.2/client64
[2022-06-22 16:59:25] NLS_LANG = AMERICAN_AMERICA.AL32UTF8
[2022-06-22 16:59:25] NLS_NCHAR = AL32UTF8
[2022-06-22 16:59:25] Trying to connect to database: dbi:Oracle:host=caldm-db2-dev;sid=KELDEV11;port=1521
[2022-06-22 16:59:27] Isolation level: SET TRANSACTION READ ONLY
[2022-06-22 16:59:27] Retrieving table information...
[2022-06-22 16:59:27] Collecting 1 tables in ALL_OBJECTS took:  0 wallclock secs ( 0.01 usr +  0.00 sys =  0.01 CPU)
[2022-06-22 16:59:27] Collecting 1 tables information in ALL_TABLES took:  0 wallclock secs ( 0.00 usr +  0.00 sys =  0.00 CPU)
[2022-06-22 16:59:27] Retrieving columns information...
[2022-06-22 16:59:28] DEBUG,
SELECT A.COLUMN_NAME, A.DATA_TYPE, A.DATA_LENGTH, A.NULLABLE, A.DATA_DEFAULT,
    A.DATA_PRECISION, A.DATA_SCALE, A.CHAR_LENGTH, A.TABLE_NAME, A.OWNER, 'NO' as "VIRTUAL_COLUMN"
FROM ALL_TAB_COLUMNS A
WHERE 1=1 AND A.OWNER='MARKH'  AND (REGEXP_LIKE(UPPER(A.TABLE_NAME), ?))
ORDER BY A.COLUMN_ID
[2022-06-22 16:59:28] Collecting 2 columns in ALL_INDEXES took:  0 wallclock secs ( 0.00 usr +  0.00 sys =  0.00 CPU)
[2022-06-22 16:59:28] [1] Scanning table MARK_TEST (1 rows)...
NOTICE:  user mapping for "kdm" already exists for server "orcl", skipping
[2022-06-22 16:59:28] Exporting tables...
[2022-06-22 16:59:28] Dumping table MARK_TEST...
DBD::Pg::db do failed: ERROR:  syntax error at or near "partition"
LINE 4:  partition varchar(200),
         ^ at /usr/local/share/perl5/Ora2Pg.pm line 8181.
[2022-06-22 16:59:28] FATAL: ERROR:  syntax error at or near "partition"
LINE 4:  partition varchar(200),
         ^, SQL:

CREATE FOREIGN TABLE ora2pg_fdw_import.mark_test (
        partition varchar(200),
        fld1 varchar(200)
) SERVER orcl OPTIONS(schema 'MARKH', table 'MARK_TEST', readonly 'true');

Aborting export...
```

What the heck am I doing wrong here ?

**Comment by darold** [comment]
Actually I was surprised to see that the ORA_RESERVED_WORDS settings was not applied anymore. Commit 7994a1e fix this issue and adds  PARTITION to the default reserved words list.

**Comment by darold** [comment]
And about the installation of ora2pg in /usr/local/bin, apparently your system wants to install the Perl modules as programs under /root/perl5/bin/ so the behavior of of the installation is correct. Instead of copying them to /usr/local/bin you might want to add /root/perl5/bin to your PATH environment variable.

**Comment by markhooper99** [comment]
Acknowledged, thanks Gilles.
Any comment about the version being reported as 23.0 rather than 23.1 ?

`Ora2Pg v23.1`     **_<-- this was reported with ora2pg -version command_**
`[2022-06-22 16:59:25] Ora2Pg version: 23.0`    **_<-- this is what it reported during an actual run_**

**Comment by darold** [comment]
You are not using the libraries in /root/perl5/lib/perl5/ but the ones in  /usr/local/share/perl5/ you should try replacing your call to ora2pg in ./marktest.sh using:
```
perl -I/root/perl5/lib/perl5 /usr/local/bin/ora2pg ...
```
I don't know why your last installation goes to /root/perl5/lib/perl5 but perhaps you are using perlbrew or there is some environment variable change that don't install the library at the default place.

**Comment by markhooper99** [comment]
Ok, after much checking I discovered the reason for the /root/perl5/... references.  It had to do with an incorrect environment setup on the 'root' user.  That's been fixed and ora2pg v23.1 re-installed.  It is now reporting the version properly when run through the marktest.sh script ...
```
hou-db6:markh:testhou> ./marktest.sh
Ora2Pg v23.1
[2022-06-23 10:08:06] Ora2Pg version: 23.1
...
```
It still fails on the 'partition' column as it did previously.  
Not sure how I download your latest change try it.

**Comment by markhooper99** [comment]
Thanks Gille !
This new version works.

Run...

```
hou-db6:markh:testhou> ./marktest.sh
Ora2Pg v23.1
[2022-06-23 12:05:18] Ora2Pg version: 23.1
[2022-06-23 12:05:18] Export type: COPY
[2022-06-23 12:05:18] Geometry export type: INTERNAL
[2022-06-23 12:05:18] ORACLE_HOME = /usr/lib/oracle/12.2/client64
[2022-06-23 12:05:18] NLS_LANG = AMERICAN_AMERICA.AL32UTF8
[2022-06-23 12:05:18] NLS_NCHAR = AL32UTF8
[2022-06-23 12:05:18] Trying to connect to database: dbi:Oracle:host=caldm-db2-dev;sid=KELDEV11;port=1521
[2022-06-23 12:05:20] Isolation level: SET TRANSACTION READ ONLY
[2022-06-23 12:05:20] Retrieving table information...
[2022-06-23 12:05:20] Collecting 1 tables in ALL_OBJECTS took:  0 wallclock secs ( 0.00 usr +  0.00 sys =  0.00 CPU)
[2022-06-23 12:05:20] Collecting 1 tables information in ALL_TABLES took:  0 wallclock secs ( 0.00 usr +  0.00 sys =  0.00 CPU)
[2022-06-23 12:05:20] Retrieving columns information...
[2022-06-23 12:05:20] Collecting column information for table ...
[2022-06-23 12:05:20] DEBUG,
SELECT A.COLUMN_NAME, A.DATA_TYPE, A.DATA_LENGTH, A.NULLABLE, A.DATA_DEFAULT,
    A.DATA_PRECISION, A.DATA_SCALE, A.CHAR_LENGTH, A.TABLE_NAME, A.OWNER
FROM ALL_TAB_COLUMNS A
WHERE 1=1 AND A.OWNER='MARKH'  AND (REGEXP_LIKE(UPPER(A.TABLE_NAME), ?))
ORDER BY A.COLUMN_ID
[2022-06-23 12:05:21] Collecting 2 columns in ALL_INDEXES took:  0 wallclock secs ( 0.00 usr +  0.00 sys =  0.00 CPU)
[2022-06-23 12:05:21] [1] Scanning table MARK_TEST (79 rows)...
[2022-06-23 12:05:21] ORACLE_HOME = /usr/lib/oracle/12.2/client64
[2022-06-23 12:05:21] NLS_LANG = AMERICAN_AMERICA.AL32UTF8
[2022-06-23 12:05:21] NLS_NCHAR = AL32UTF8
[2022-06-23 12:05:21] Trying to connect to database: dbi:Oracle:host=caldm-db2-dev;sid=KELDEV11;port=1521
[2022-06-23 12:05:22] Exporting tables...
[2022-06-23 12:05:22] Dumping table MARK_TEST...
[2022-06-23 12:05:22] ORACLE_HOME = /usr/lib/oracle/12.2/client64
[2022-06-23 12:05:22] NLS_LANG = AMERICAN_AMERICA.AL32UTF8
[2022-06-23 12:05:22] NLS_NCHAR = AL32UTF8
[2022-06-23 12:05:22] Trying to connect to database: dbi:Oracle:host=caldm-db2-dev;sid=KELDEV11;port=1521
[2022-06-23 12:05:24] Isolation level: SET TRANSACTION READ ONLY
[2022-06-23 12:05:25] Exporting data of table MARK_TEST using foreign table...
[2022-06-23 12:05:25] Setting search_path using: SET search_path = markh,public,public;
...
[2022-06-23 12:05:25] Truncating table mark_test...
[2022-06-23 12:05:25] Exporting foreign table data for MARK_TEST using query: INSERT INTO mark_test ("partition",fld1) OVERRIDING SYSTEM VALUE SELECT "partition",fld1 FROM ora2pg_fdw_import."mark_test"
[2022-06-23 12:05:27] ORACLE_HOME = /usr/lib/oracle/12.2/client64
[2022-06-23 12:05:27] NLS_LANG = AMERICAN_AMERICA.AL32UTF8
[2022-06-23 12:05:27] NLS_NCHAR = AL32UTF8
[2022-06-23 12:05:27] Trying to connect to database: dbi:Oracle:host=caldm-db2-dev;sid=KELDEV11;port=1521
[2022-06-23 12:05:29] Isolation level: SET TRANSACTION READ ONLY
[2022-06-23 12:05:30] Restarting sequences
[2022-06-23 12:05:30] Extracted sequence information for sequence "TSM_ID_SEQ"
[2022-06-23 12:05:30] Extracted sequence information for sequence "SSIP_ID_SEQ"
[2022-06-23 12:05:30] Extracted sequence information for sequence "WORLD_STATES_BOARDER_SEQ"
NOTICE:  relation "tsm_id_seq" does not exist, skipping
NOTICE:  relation "ssip_id_seq" does not exist, skipping
NOTICE:  relation "world_states_boarder_seq" does not exist, skipping
[2022-06-23 12:05:30] Total time to export data from 1 tables (0 partitions, 0 sub-partitions) and 79 total rows:  8 wallclock secs ( 0.87 usr  0.27 sys +  0.02 cusr  0.04 csys =  1.20 CPU)
```
